src/odemis/gui/comp/miccanvas.py

```diff
@@ -1250,32 +1250,38 @@ def get_hcontent_range(self):
         """
         To allow fitting to content, detect where the first leftmost non-zero value
         occurs, and also detect where the first rightmost non-zero value occurs.
-        Returns the horizontal range that fits the data content
-        If no range is found, return none
+        return ((float, float) or None): the horizontal range that fits the data
+          content. If no range is found, return None.
         """
         if self._data_buffer is None:
             return None
 
-        (xd , yd) = self._data_buffer
-        # If the empty portion on the edge is less than the threshold percent of
-        # the full data width, do not include it.
-        threshold = 0.05
+        xd, yd = self._data_buffer
 
+        ileft = 0
```

Comment:
can you add a comment here, explaining why this is necessary?

Reply:
I believe this is in case len(yd) == 0, in which case ileft is never set. Probably a comment about this would be nice though.